great_expectations/datasource/fluent/interfaces.py

```diff
@@ -1243,3 +1257,123 @@ def _create_validator(self, *, result_format: ResultFormatUnion) -> V1Validator:
             batch_parameters=self.batch_request.options,
             result_format=result_format,
         )
+
+    @overload
+    def compute_metrics(self, metrics: Metric) -> MetricResult: ...
+
+    @overload
+    def compute_metrics(self, metrics: list[Metric]) -> list[MetricResult]: ...
+
+    def compute_metrics(self, metrics: Metric | list[Metric]) -> MetricResult | list[MetricResult]:
+        """Compute one or more metrics on this Batch.
+
+        Args:
+            metrics: A single Metric or list of Metrics to compute.
+                     Each Metric must be an instance of a concrete Metric subclass.
+
+        Returns:
+            If a single Metric is provided, returns a single MetricResult.
+            If a list of Metrics is provided, returns a list of MetricResults,
+            in the same order as the input metrics were received.
+            For metrics without a defined MetricResult generic type,
+            the base MetricResult class will be returned.
+
+        Examples:
+            >>> batch.compute_metrics(BatchRowCount())
+            BatchRowCountResult(id=..., value=1000)
+
+            >>> batch.compute_metrics([
+            ...     BatchRowCount(),
+            ...     ColumnMax(column="age")
+            ... ])
+            [BatchRowCountResult(id=..., value=1000), ColumnMaxResult(id=..., value=85)]
+
+        Notes:
+            Until this mypy bug is resolved, lists of Metrics are being incorrectly
+            inferred by the static type checker as list[Domain]. You can work around
+            this by adding an explicit annotation (e.g. metrics: list[Metric] = ...)
+            https://github.com/python/mypy/issues/18712
+        """
+        if isinstance(metrics, Metric):
+            metrics = [metrics]
+
+        execution_engine = self.data.execution_engine
+        execution_engine.batch_manager.load_batch_list(batch_list=[self])
+        metrics_calculator = MetricsCalculator(execution_engine=execution_engine)
```

Comment:
This preamble necessary to setup the metrics calculator is used 2 other places in this class and takes real domain knowledge to understand that one needs to do this. Can you put this in a private helper method, eg `_get_metrics_calculator()` and use it here and the other 2 places.